File: ruby/muse/util/RandomFactory.rb
require "securerandom"

module RandomFactory
  def self.genIntN(min, max)
    min + SecureRandom.random_number(max - min + 1)
  end

  def self.genInt
    genIntN(1, 2_147_483_647)
  end

  def self.genEven
    genInt >> 1 << 1
  end

  def self.genOdd
    genInt | 1
  end
end

File: php/muse/util/RandomFactory.php
<?php

declare(strict_types=1);

namespace muse\util\RandomFactory;

function genIntN(int $min, int $max): int
{
    return random_int($min, $max);
}

function genInt(): int
{
    return genIntN(1, 2147483647);
}

function genEven(): int
{
    return genInt() >> 1 << 1;
}

function genOdd(): int
{
    return genInt() | 1;
}
